src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -66,7 +104,17 @@ protected override string GetDatabaseFilePath(string workingFolderPath)
             }
             catch (Exception)
             {
-                sqlStorage?.Dispose();
+                if (sqlStorage != null)
```

Comment:
if exception was thrown, shouldn't we always dispose the owner lock?

Reply:
ah. just realized that the dispose of the storage will dispose the lock. sorry about that.